opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/Propagation.java

```diff
@@ -73,7 +73,12 @@ private TextMapGetter() {}
 
     @Override
     public String get(Map<String, String> carrier, String key) {
-      return carrier.get(key);
+      for (Map.Entry<String, String> entry : carrier.entrySet()) {
```

Comment:
If we plan to iterate over all the values per `get()` call, we should get rid of the intermediate temporary `Map`.

Reply:
I didn't remove temporary map because we need to implement 
```
 @Override
public Iterable<String> keys(Map<String, String> carrier) {
      return carrier.keySet();
}
```

if carrier will be `TextMapExtract` then to get keys I will need to create something temporal...